app/src/main/java/com/hedvig/app/feature/keygear/ui/ReceiptActivity.kt

```diff
@@ -1,37 +1,126 @@
 package com.hedvig.app.feature.keygear.ui
 
 import android.content.Intent
+import android.graphics.Bitmap
+import android.graphics.drawable.Drawable
 import android.os.Bundle
+import android.os.Environment
+import android.os.Parcelable
+import androidx.core.content.FileProvider
 import com.bumptech.glide.Glide
+import com.bumptech.glide.load.resource.bitmap.CenterCrop
+import com.bumptech.glide.request.target.CustomTarget
+import com.bumptech.glide.request.transition.Transition
 import com.hedvig.app.BaseActivity
 import com.hedvig.app.R
+import com.hedvig.app.util.extensions.view.setHapticClickListener
+import kotlinx.android.parcel.Parcelize
 import kotlinx.android.synthetic.main.activity_receipt.*
+import java.io.File
+import java.io.FileOutputStream
 
-class ReceiptActivity : BaseActivity() {
+class ReceiptActivity : BaseActivity(R.layout.activity_receipt) {
+
+    private var tempPhotoPath = ""
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_receipt)
 
-        val url = "https://upload.wikimedia.org/wikipedia/commons/0/0b/ReceiptSwiss.jpg"
+        val intent = intent
+        val data = intent.getParcelableExtra<Receipt>("RECEIPT_DATA")
+
+        val fileUrl = data.file.key
+
+        Glide.with(this)
```

Comment:
Can we do this when the user clicks the share-button?

By not having this run in `onCreate()` we remove the need for the `tempPhotoPath`-variable and avoid creating files until the user needs the file to exist.